src/Formats/TemporaryFileStream.cpp

```diff
@@ -39,6 +39,10 @@ void TemporaryFileStream::write(const std::string & path, const Block & header,
         output.write(block);
 
     compressed_buf.finalize();
+    output.flush();
+    file_buf.next();
+
+    return Stat{compressed_buf.count(), file_buf.count()};
```

Comment:
Why not use `CompressedWriteBuffer` methods `getCompressedBytes` and `getUncompressedBytes`?
This really confused me, wasn't sure if you switched them around but it turned out correct.

Reply:
Good idea, updated BufferingToFileTransform and TemporaryFileStream to use this methods